driver-dse/src/main/java/com/datastax/driver/graph/BoundGraphStatement.java

```diff
@@ -0,0 +1,116 @@
+import com.datastax.driver.core.exceptions.DriverException;
+
+/**
+ * Bound graph statement produced from a prepared statement.
+ */
+public class BoundGraphStatement extends AbstractGraphStatement<BoundStatement> {
+
+    //  "name", "value"
+    Map<String, Object> valuesMap;
+
+    BoundGraphStatement(BoundStatement bs, GraphStatement gst) {
+        super(gst.getSession());
+        this.wrappedStatement = bs;
+        this.valuesMap = new HashMap<String, Object>();
+
+        /*
+        Need to keep the configuration from the Prepared statement to
+        the created BoundStatement, so users don't have to re configure the payload and such.
+         */
+        copyConfigFromStatement(gst);
+    }
+
+    void copyConfigFromStatement(GraphStatement gst) {
+        this.wrappedStatement.setOutgoingPayload(gst.getGraphOptions() == null ? this.session.getDefaultGraphOptions() : gst.getGraphOptions());
+    }
+
+    // Bind variables in the PreparedStatement
+    @Override
+    BoundStatement configureAndGetWrappedStatement() {
+        JsonNodeFactory factory = new JsonNodeFactory(false);
+        JsonFactory jsonFactory = new JsonFactory();
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            int paramNumber = 0;
+            for (Map.Entry<String, Object> param : this.valuesMap.entrySet()) {
+                StringWriter stringWriter = new StringWriter();
+                JsonGenerator generator = jsonFactory.createGenerator(stringWriter);
+                ObjectNode parameter = factory.objectNode();
+                String name = param.getKey();
+                Object value = param.getValue();
+                if (value instanceof Integer) {
+                    parameter.put("name", name);
+                    parameter.put("value", (Integer)value);
+                } else if (value instanceof String) {
+                    parameter.put("name", name);
+                    parameter.put("value", (String)value);
+                } else if (value instanceof Float) {
+                    parameter.put("name", name);
+                    parameter.put("value", (Float)value);
+                } else if (value instanceof Double) {
+                    parameter.put("name", name);
+                    parameter.put("value", (Double)value);
+                } else if (value instanceof Boolean) {
+                    parameter.put("name", name);
+                    parameter.put("value", (Boolean)value);
+                } else if (value instanceof Long) {
+                    parameter.put("name", name);
+                    parameter.put("value", (Long)value);
+                } else {
+                    throw new DriverException("Parameter : " + value + ", is not in a valid format to be sent as Gremlin parameter.");
+                }
+                objectMapper.writeTree(generator, parameter);
+                this.wrappedStatement.setString(paramNumber++, stringWriter.toString());
```

Comment:
This was made to follow the [specs](https://github.com/riptano/bdp/blob/DSP-5175/dse-graph/docs/dev/server.adoc#dse-graph-server-extension), parameters are sent as unnamed parameters, each containing `{name:'parameter_name', value:JSON_serialised_value}`. 
I think this is related to the way the tinkerpop's gremlin server takes parameters, and since the DSE server will embed a gremlin-server, they chose this format to avoid modifiying the embedded gremlin server.